src/server/jobs/update-bcast.js

```diff
@@ -0,0 +1,271 @@
+    url: `${process.env.BRUINCAST_API}?display_id=ccle_api_courses&args[0]=${term}`,
+    headers: {
+      Cookie: bcastCookie,
+    },
+  };
+
+  await axios(config)
+    .then(async function(response) {
+      await callback(JSON.stringify(response.data));
+    })
+    .catch(error => {
+      throw error;
+    });
+}
+
+/**
+ * Sends GET request to BruinCast API to get media for given term and Class ID.
+ * @param  {String} term Term with spelled out quarter and full year
+ * @param  {String} classID Nine-digit Class ID (a.k.a. SRS number)
+ * @param  {String} callback Name of callback function
+ */
+async function getMedia(term, classID, callback) {
+  const config = {
+    method: 'get',
+    url: `${process.env.BRUINCAST_API}?display_id=ccle_api_media&args[0]=${term}&args[1]=${classID}`,
+    headers: {
+      Cookie: bcastCookie,
+    },
+  };
+
+  await axios(config)
+    .then(async function(response) {
+      await callback(JSON.stringify(response.data));
+    })
+    .catch(error => {
+      throw error;
+    });
+}
+
+/**
+ * Inserts an array of JSON-formatted entries into database's BruinCast collection.
+ * @param  {MongoClient} client A connected MongoClient
+ * @param  {ClientSession} session A session started by client
+ * @param  {[Object]} entries An array of JSON-formatted objects for media entries
+ */
+async function insertMediaEntries(client, session, entries) {
+  // logger.info(entries);
+  const result = await client
+    .db(process.env.DB_DATABASE)
+    .collection('bruincastmedia')
+    .insertMany(entries, { session });
+  logger.info(`Added ${result.insertedCount} record(s).`);
+}
+
+/**
+ * Deletes all media entries for a given term and Class ID
+ * @param  {MongoClient} client A connected MongoClient
+ * @param  {ClientSession} session A session started by client
+ * @param  {String} classTerm A class term formatted as YYQ
+ * @param  {String} classIDnum A 9-digit Class ID
+ */
+async function deleteMediaEntriesForClass(
+  client,
+  session,
+  classTerm,
+  classIDnum
+) {
+  const query = {
+    term: classTerm,
+    classID: classIDnum,
+  };
+  const result = await client
+    .db(process.env.DB_DATABASE)
+    .collection('bruincastmedia')
+    .deleteMany(query, { session });
+  logger.info(`Deleted ${result.deletedCount} record(s).`);
+}
+
+/**
+ * For a given term and Class ID, delete all records and insert new entries
+ * @param  {MongoClient} client A connected MongoClient
+ * @param  {String} classTerm A class term formatted as YYQ
+ * @param  {String} classIDnum A 9-digit Class ID
+ * @param  {[Object]} entries An array of JSON-formatted objects for media entries
+ */
+async function updateRecordsForClass(client, classTerm, classIDnum, entries) {
+  const session = client.startSession();
+  session.startTransaction();
+  try {
+    await deleteMediaEntriesForClass(client, session, classTerm, classIDnum);
+    await insertMediaEntries(client, session, entries);
+    await session.commitTransaction();
```

Comment:
@chern I noticed when running the script that a lot of noise is in the logs. It always says deleted 20 record(s) then added 20 record(s). Can you just log the net differences? So it should just say Added 1 record, or Deleted 2 records, just the net value.